sdk/keyvault/keyvault/CMakeLists.txt

```diff
@@ -8,25 +8,15 @@ project (az_keyvault LANGUAGES C)
 set(CMAKE_C_STANDARD 99)
 
 add_library (
-  az_keyvault STATIC
+  az_keyvault
   src/az_keyvault_client.c
   )
 
 target_include_directories (az_keyvault PUBLIC inc)
 target_include_directories (az_keyvault PRIVATE src)
 
-target_link_libraries(az_keyvault PUBLIC az_core)
-target_link_libraries(az_keyvault PRIVATE az_core_internal)
```

Comment:
I know what you are trying to achieve here, but try to keep az_core and az_core_internal, it worked for me for the PAL layer, give it a try

Reply:
can't, target_link_libs require a path, if I add the target, we got the lib and not the headers.
Here we only are interested in header for function definitions